src/lib/backup.ts

```diff
@@ -1,40 +1,168 @@
 import { PendingReverseSwap, PendingSubmarineSwap } from '@arkade-os/boltz-swap'
-import { ContractRepositoryImpl } from '@arkade-os/sdk'
 import { IndexedDBStorageAdapter } from '@arkade-os/sdk/adapters/indexedDB'
-import { NostrStorageData, NostrStorage, nostrAppName } from './nostr'
+import { ContractRepositoryImpl } from '@arkade-os/sdk'
+import { getPublicKey } from 'nostr-tools/pure'
+import { NostrStorage } from './nostr'
 import { Config } from './types'
+import { consoleError } from './logs'
 
 const storage = new IndexedDBStorageAdapter('arkade-service-worker')
 const contractRepo = new ContractRepositoryImpl(storage)
 
-export const handleNostrBackup = async (config: Config) => {
-  // data to backup
-  const backupData: NostrStorageData = {
-    config,
-    reverseSwaps: (await contractRepo.getContractCollection('reverseSwaps')) as PendingReverseSwap[],
-    submarineSwaps: (await contractRepo.getContractCollection('submarineSwaps')) as PendingSubmarineSwap[],
-  }
-  // save to nostr
-  const nostrStorage = new NostrStorage({ pubkey: config.pubkey })
-  await nostrStorage.save(nostrAppName, backupData)
+type NostrStorageData = {
+  config?: Config
+  reverseSwaps?: PendingReverseSwap[]
+  submarineSwaps?: PendingSubmarineSwap[]
 }
+export class BackupProvider {
+  private nostrStorage: NostrStorage
+  private seckey: Uint8Array | null
+  private pubkey: string
+
+  /**
+   * Initialize Backup with either a secret key or public key
+   * @param options.seckey - Optional secret key (Uint8Array). If provided, pubkey is derived.
+   * @param options.pubkey - Optional public key (hex string). Required if seckey not provided.
+   * @throws Error if neither seckey nor pubkey is provided, or if pubkey format is invalid
+   */
+  constructor(options: { pubkey?: string; seckey?: Uint8Array }) {
```

Comment:
now I understand why we need https://github.com/arkade-os/ts-sdk/pull/223
we need to think about it